src/__tests__/Search/SearchView.test.tsx

```diff
@@ -152,79 +140,78 @@ describe('Base Search', () => {
     const comment = await screen.findAllByText("you've got no arms left!");
     expect(comment[0]).toBeInTheDocument();
 
-    expect(store.getState().search[searchType].searchResults).toStrictEqual(
-      testData,
-    );
-
     // Press Escape key to hide search results.
 
     await user.keyboard('{Escape}');
     expect(comment[0]).not.toBeInTheDocument();
   });
 
   it('Should not call fetch if search value is not a hash or email', async () => {
-    const spyOnFetch = jest.spyOn(global, 'fetch');
     // set delay to null to prevent test time-out due to useFakeTimers
     const user = userEvent.setup({ delay: null });
     renderComponent();
 
     const searchInput = screen.getAllByRole('textbox')[0];
 
+    // We're running fake timers after each user action, because the input
+    // normally waits 500ms before doing requests. Because we want to test the
+    // requests, we have to run the timers to properly assess the result.
+    // They're all wrapped in "act" because they might also trigger state
+    // changes and rerenders.
```

Comment:
Do we have to run all the timers post all user actions in our test. I understand why but I'm wondering if the above is an example or a standard we want to follow with current and future tests?

Reply:
not always and all the time
but _in this case_ we need to run them so that the action happening after 500ms is actually triggered.

Another way would be to use `screen.findByXXX`, for example we could wait for the popup to be displayed, and then for the popup to disappear. I didn't think of that when I wrote it initially.
`screen.findByXXX` uses `act` already so we wouldn't need to use it explicitely ourselves, and it also runs fake timers automatically.
It's also closer to what a normal user would do (do an action, then wait for something happening).

Let me try to do it this way and if I can make it work easily I'll land it. Otherwise I'll keep it this way. What do you think?

Reply:
I think for the sake of time you can land it and we can update later. But the latter is a good idea re `screen.findByXXX`.